sound/soc/sof/sof-pci-dev.c

```diff
@@ -277,14 +258,11 @@ static int sof_pci_probe(struct pci_dev *pci,
 	dev_dbg(dev, "created machine %s\n",
 		dev_name(&sof_pdata->pdev_mach->dev));
 
-	/* continue probing after firmware is loaded */
-	dev_info(dev, "info: loading firmware %s\n", mach->sof_fw_filename);
-	ret = request_firmware_nowait(THIS_MODULE, true, mach->sof_fw_filename,
-				      dev, GFP_KERNEL, priv, sof_pci_fw_cb);
+	/* register sof-audio platform driver */
+	ret = sof_create_platform_device(priv);
 	if (ret) {
 		platform_device_unregister(sof_pdata->pdev_mach);
-		dev_err(dev, "error: failed to load firmware %s\n",
-			mach->sof_fw_filename);
+		dev_err(dev, "error: failed to create platform device!\n");
 		goto release_regions;
```

Comment:
here it's different from api, the runtime_pm is skipped.

Reply:
this is correct.